### PR TITLE
Add logic for parting a TOTP code from a Uri or Intent

### PR DESCRIPTION
**Pull request by david-livefront**
## 🎟️ Tracking

N/A

## 📔 Objective

This PR adds logic for parsing a TOTP uri. This logic is not used yet but will be in a forthcoming PR.

## ⏰ Reminders before review

- Contributor guidelines followed
- All formatters and local linters executed and passed
- Written new unit and / or integration tests where applicable
- Used internationalization (i18n) for all UI strings
- CI builds passed
- Communicated to DevOps any deployment requirements
- Updated any necessary documentation or informed the documentation team

## 🦮 Reviewer guidelines

- 👍 (`:+1:`) or similar for great changes
- 📝 (`:memo:`) or ℹ️ (`:information_source:`) for notes or general info
- ❓ (`:question:`) for questions
- 🤔 (`:thinking:`) or 💭 (`:thought_balloon:`) for more open inquiry that's not quite a confirmed
  issue and could potentially benefit from discussion
- 🎨 (`:art:`) for suggestions / improvements
- ❌ (`:x:`) or ⚠️ (`:warning:`) for more significant problems or concerns needing attention
- 🌱 (`:seedling:`) or ♻️ (`:recycle:`) for future improvements or indications of technical debt
- ⛏ (`:pick:`) for minor or nitpick changes

### CODE REVIEW

File: app/src/main/java/com/x8bit/bitwarden/ui/vault/model/TotpData.kt [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+) : Parcelable {
+    /**
+     * A representation of the various cryptographic hash algorithms used by TOTP.
+     */
+    enum class CryptoHashAlgorithm(val value: String) {
+        SHA_1(value = "sha1"),
+        SHA_256(value = "sha256"),
+        SHA_512(value = "sha512"),
+        MD_5(value = "md5"),
+        ;
```
Comment (ahaisting-livefront) [inline app/src/main/java/com/x8bit/bitwarden/ui/vault/model/TotpData.kt:39]: ⛏️ Is this supposed to be here?

**Review by ahaisting-livefront — APPROVED** [review]
Looks good. Nice tests too!

File: app/src/main/java/com/x8bit/bitwarden/ui/vault/model/TotpData.kt (hunk 1 above)
Comment (david-livefront) [reply app/src/main/java/com/x8bit/bitwarden/ui/vault/model/TotpData.kt:39]: Yea, you need a trailing semi-colon if you want to add a companion object

**Comment by david-livefront** [comment]
Thanks @ahaisting-livefront

**Pull request merged**